.github/workflows/test.yml

```diff
@@ -0,0 +1,60 @@
+---
+name: Test
+
+on:
+  pull_request:
+  push:
+    branches:
+      - 'develop'
+      - '*-stable'
+concurrency:
+  group: ${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  rubocop:
+    name: Rubocop
+    uses: theforeman/actions/.github/workflows/rubocop.yml@v0
+
+  test:
+    name: "Ruby ${{ matrix.ruby }} / Puppet ${{ matrix.puppet }}"
+    needs: rubocop
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - ruby: '2.7'
+            puppet: '7.0'
+          - ruby: '3.0'
+            puppet: '7.0'
+          - ruby: '3.2'
+            puppet: '8.0'
```

Comment:
I copied this from kafo and do not remember why we had added 3/7 here, but I am sure there was a reason :D

Reply:
https://github.com/theforeman/foreman-installer/pull/942 would be an attempt at that.